Flaky test: `TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers`

```
--- FAIL: TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers (3.88s)
    --- FAIL: TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers/RF_=_2 (1.49s)
        multitenant_test.go:1617: 
            	Error Trace:	multitenant_test.go:1617
            	Error:      	Not equal: 
            	            	expected: 1
            	            	actual  : 0
            	Test:       	TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers/RF_=_2
FAIL
```

--- a/pkg/alertmanager/multitenant_test.go
+++ b/pkg/alertmanager/multitenant_test.go
@@ -1820,7 +1820,14 @@ func TestAlertmanager_StateReplicationWithSharding_InitialSyncFromPeers(t *testi
 			{
 				metrics := registries.BuildMetricFamiliesPerUser()
 				assert.Equal(t, float64(1), metrics.GetSumOfGauges("cortex_alertmanager_silences"))
-				assert.Equal(t, float64(1), metrics.GetSumOfCounters("cortex_alertmanager_state_replication_total"))
+			}
+			// 2.c. Wait for the silence replication to be attempted; note this is asynchronous.
+			{
+				test.Poll(t, 5*time.Second, float64(1), func() interface{} {
+					metrics := registries.BuildMetricFamiliesPerUser()
+					return metrics.GetSumOfCounters("cortex_alertmanager_state_replication_total")
+				})
+				metrics := registries.BuildMetricFamiliesPerUser()
 				assert.Equal(t, float64(0), metrics.GetSumOfCounters("cortex_alertmanager_state_replication_failed_total"))
 			}
 

